### ISSUE TITLE
Align the default link states with Gov

### ISSUE THREAD

**Issue by paulfourniercarey** [nhsuk/nhsuk-service-manual-community-backlog#519]
## What
Gov reworked their link states to make them more readable, it makes sense to align with this unless we have research that can show it work better as it is.

## Why
It better aligns with Gov and is often seen in the wild too.
Blog post from 2021
https://designnotes.blog.gov.uk/2021/07/07/making-links-easier-to-see-and-read-on-gov-uk/

**Comment by frankieroberto** [comment]
@paulfourniercarey thanks for adding! This seems like a good thing to port across from the GOV.UK Design System to me. @anandamaryon1 agree?

**Comment by anandamaryon1** [comment]
I agree this is worth exploring. Will need some testing, since we use a different font and hover state.

**Comment by anandamaryon1** [comment]
We now have an open PR for this: https://github.com/nhsuk/nhsuk-frontend/pull/1366

We're now trying to settle on the finer detail, which needs to be tweaked for our font, which has lower descenders than GOV's Transport font. 

**Current styling (without the enhanced underlines):**
![Image](https://github.com/user-attachments/assets/acb6214b-045a-4f6a-8f7a-dd204239c9c6)

**As ported from GOV's implementation:**
![Image](https://github.com/user-attachments/assets/6c10fbdd-9615-409e-8974-d7504daffe1a)

**With slightly increased underline offset**
![Image](https://github.com/user-attachments/assets/0cad7a3c-c139-4757-9599-bb06dc425a40)

**With slightly increased underline offset and ink-skip removed:**
![Image](https://github.com/user-attachments/assets/444aff13-f9a5-4baa-aebd-2c428de0b7c4)

My take is that increasing the offset and removing the ink-skip (little gaps in the underline for descenders) helps with legibility, but interested to hear more opinions.

**Comment by michaelgallagher** [comment]
v2 for me ("As ported from GOV's implementation")

**Comment by Arpana18** [comment]
can't make out the difference between V3 (With slightly increased underline offset) and V4 (With slightly increased underline offset and ink-skip removed). I liked the way offset is increased, that lets the descenders fall within the underline. 

however an observation for V3: on page zoom, the underline breaks for the descenders. don't know if it it deliberate or a defect.

**Comment by frankieroberto** [comment]
V2 for me - feels a bit odd to have the underline entirely below but touching the descenders? Marginal though.

I'm also not 100% sure how consistent this is across browsers (particularly the ink-skipping), so we might have to live with some small amount of variance?
